geode-junit/src/test/java/org/apache/geode/test/junit/rules/ConcurrencyRuleTest.java

```diff
@@ -201,6 +250,21 @@ public void runAndExpectValue_failsForWrongValue(Execution execution) {
     assertThat(invoked.get()).isTrue();
   }
 
+  @Test
+  public void failsWhenMultipleReturnValuesExpected() {
```

Comment:
This only tests one of the `if (outcomeSet)` conditionals. For complete unit test coverage there should be an additional test for each of the  conditionals.